strawberry/annotation.py

```diff
@@ -182,11 +187,61 @@ def create_union(self, evaled_type: Type) -> StrawberryUnion:
             return evaled_type
 
         types = evaled_type.__args__
+
         union = StrawberryUnion(
             type_annotations=tuple(StrawberryAnnotation(type_) for type_ in types),
         )
+
+        union_args = [arg for arg in args if isinstance(arg, StrawberryUnion)]
+        if len(union_args) > 1:
+            logging.warning(
+                "Duplicate union definition detected. "
+                "Only the first definition will be considered"
+            )
+
+        if union_args:
+            arg = union_args[0]
+            union.graphql_name = arg.graphql_name
+            union.description = arg.description
+            union.directives = arg.directives
+
+            union._source_file = arg._source_file
+            union._source_line = arg._source_line
+
+        self.validate_union_members(types, union)
+
         return union
 
+    def validate_union_members(
+        self, types: Tuple[type], union: StrawberryUnion
+    ) -> None:
```

Comment:
Since this doesn't use `self`, I wonder if it would be better to define it as a separated function.

Even better, define it as a method in `StrawberryUnion` as `def validate_members(types: Tuple[type])`, or even without `types`, since the union already have those inside of it.